### PR TITLE
pkp/pkp-lib#9195 Added queue jobs control configs

### PR DESCRIPTION
**Pull request by touhidurabir**
for pkp/pkp-lib#9195

### CODE REVIEW

File: jobs/BaseJob.php [hunk 1]
```diff
@@ -47,12 +47,53 @@ abstract class BaseJob implements ShouldQueue
...
+     * The number of times the job may be attempted.
+     *
+     * @var int
+     */
+    public $tries = 3;
+
+    /**
+     * The number of SECONDS to wait before retrying the job.
+     *
+     * @var int
```
Comment (asmecher) [inline jobs/BaseJob.php:60]: Rather than `@var` self-documentation, better to just use the `public int $backoff = 5;` directly. (Also below)

File: jobs/BaseJob.php [hunk 2]
```diff
@@ -47,12 +47,53 @@ abstract class BaseJob implements ShouldQueue
...
+
+    /**
+     * Indicate if the job should be marked as failed on timeout.
+     *
+     * @var bool
+     */
+    public $failOnTimeout = true;
+
+    /**
+     * Initiate the job
```
Comment (asmecher) [inline jobs/BaseJob.php:86]: Should be `Initialize` rather than `Initiate`, I think -- `Initiate` implies that calling the constructor starts the job. (Also below)

File: jobs/email/EditorialReminder.php [hunk 3]
```diff
@@ -81,7 +81,7 @@ public function handle(): void
 
         /** @var int $submissionId */
         foreach ($submissionIds as $submissionId) {
-            $submission = Repo::submission()->get($submissionId);
+            $submission = Repo::submission()->get((int)$submissionId);
```
Comment (asmecher) [inline jobs/email/EditorialReminder.php:84]: Is this change necessary?

File: jobs/email/EditorialReminder.php [hunk 4]
```diff
@@ -81,7 +81,7 @@ public function handle(): void
...
             $submissions[$submissionId] = $submission;
 
             if ($submission->getData('stageId') === WORKFLOW_STAGE_ID_SUBMISSION) {
```
Comment (asmecher) [inline jobs/email/EditorialReminder.php:87]: This could be one of those weird data type issues, I think. `$submission->getData('stageId')` might come back from the DB as a string. To be safe, use a `==` comparison.

**Review by asmecher — COMMENTED** [review]
Thanks, @touhidurabir! Am I reading correctly that this adds the capability for additional configurations but doesn't use it anywhere? I'm not a big fan of adding capabilities unless we at least demonstrate their use somewhere in the codebase; otherwise it's dead code. Is there a case or two where this would prove useful to end users that we could add to the PR?

**Comment by touhidurabir** [comment]
> this adds the capability for additional configurations but doesn't use it anywhere

@asmecher not the case . once these added , this will have reflect on both case when job run via `JobRunner` or from `CLI` or `worker` . And these add few enhancement like specially the `backOff` property that can have impact when failed job picked too fast to execute again . The other few just add some extra enhancement just to make sure it a bit higher threshold than default like the `timeout` , `maxException` and `failOnTimeout` . 

for example, `timeout` will be helpful for long running jobs such as balk mailing which may need more than 60 seconds which is default . The same goes for `failOnTimeout` as it's right to make a job marked as failed when passed the timeout limit before complete and exit .

File: jobs/email/EditorialReminder.php (hunk 3 above)
Comment (touhidurabir) [reply jobs/email/EditorialReminder.php:84]: @asmecher this is added because of this error posted in forum https://forum.pkp.sfu.ca/t/ojs-3-4-0-1-failed-jobs/80170/4?u=touhidur . I also could not reproduce it in my testing but may be this is one of those platform specific type issues that we are experiencing too randomly .

File: jobs/email/EditorialReminder.php (hunk 4 above)
Comment (touhidurabir) [reply jobs/email/EditorialReminder.php]: Done .

File: jobs/BaseJob.php (hunk 2 above)
Comment (touhidurabir) [reply jobs/BaseJob.php]: done .

File: jobs/BaseJob.php (hunk 1 above)
Comment (touhidurabir) [reply jobs/BaseJob.php]: Done .

File: jobs/email/EditorialReminder.php (hunk 3 above)
Comment (asmecher) [reply jobs/email/EditorialReminder.php:84]: I think this is only happening because you've got [strict types](https://github.com/pkp/pkp-lib/pull/9213/files#diff-83d61d197c61d91e16b32cc318edb9f2c3ba21032384d2337d75ce12c7189fdfR3) enabled on the file; otherwise the `string` would be automatically coerced to an `int` when passed into the submission repository's `get` function.

We don't yet have a set of development patterns that will gracefully allow us to adopt strict typing -- so rather than peppering the code with manual typecasts when errors arise, I'd rather just disable strict typing.

**Comment by asmecher** [comment]
Thanks, @touhidurabir! I just added [one last comment](https://github.com/pkp/pkp-lib/pull/9213#discussion_r1295046909); please let me know when you're ready for me to take another look.

**Comment by touhidurabir** [comment]
@asmecher please take another look. if all ok, should be good to merge .
